cpp/include/raft/core/interruptible.hpp

```diff
@@ -203,21 +205,25 @@ class interruptible {
   {
     std::lock_guard<std::mutex> guard_get(mutex_);
     // the following constructs an empty shared_ptr if the key does not exist.
-    auto& weak_store  = registry_[thread_id];
+    auto& weak_store  = (*registry_)[thread_id];
```

Comment:
NB: since we know the registry can only be deleted on program exit, accessing it here without checks shouldn't cause any problems.

Reply:
Thanks for this fix, @achirkin! Does the reproducer in #1225 also no longer segfault with this change?

Reply:
At least on my machine, yes!